### ISSUE TITLE
jq prints nothing and exits

### ISSUE THREAD

**Issue by MatrixManAtYrService** [stedolan/jq#1967]
**Describe the bug**
Certain input strings cause jq to exit with code zero and print nothing.

**To Reproduce**

```
❯ echo '{"foo":{"bar":{"baz":{"qux":{"quux1":{"ping":"pong","quux2":{}}}},"oof":{}},"rab":"zab"}' | jq . 
```

**Expected behavior**
The json is malformed, so an error should be reported to the user

**Environment**
```
❯ uname -a

    Linux somehostname 5.0.0-25-generic #26~18.04.1-Ubuntu SMP Thu Aug 1 13:51:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

❯ jq --version

    jq-1.5-1-a5b5cbe
```

**Details**

Here's a fluffed up version of the input string:
```
{
  "foo": {
    "bar": {
      "baz": {
        "qux": {
          "quux1" : {
              "ping" : "pong",
          "quux2": {
          }
        }
      }
    },
    "oof": {}
  },
  "rab": "zab"
}
```
python -m json.tool says this about it:
```
Expecting object: line 16 column 2 (char 204)                                                                                 
```
The modification below causes the json to parse and reformat correctly, but the bug is that the unmodified version doesn't print an error.
```
 "ping" : "pong", -->  "ping" : "pong" },
```

**Comment by pkoppstein** [comment]
I think you'll find this type of problem has been addressed in jq version 1.6:

```
echo '{"foo":{"bar":{"baz":{"qux":{"quux1":{"ping":"pong","quux2":{}}}},"oof":{}},"rab":"zab"}' | jq-1.6 . 
parse error: Unfinished JSON term at EOF at line 2, column 0
```

**Comment by MatrixManAtYrService** [comment]
@pkoppstein thanks for pointing that out.  It might be time for me to switch to a distro with more up to date packages.
